src/account.ts

```diff
@@ -4,154 +4,198 @@ import {
-            account: '............1',
-            authorized_by: '............1',
-            permission: permissionName,
+    // TODO: Refactor once resources library is updated
+    resources(sampleAccount?: NameType) {
+        // Returns an instance of the @wharfkit/resources library
+        //   configured for this blockchain/account
+        return new Resources({
+            api: this.client,
+            sampleAccount: sampleAccount ? String(sampleAccount) : undefined,
+            symbol: this.data.core_liquid_balance
+                ? String(this.data.core_liquid_balance.symbol)
+                : undefined,
         })
     }
 
-    buyRam(amount: AssetType): Action {
-        return this.eosioContract.action('buyram', {
-            payer: '............1',
-            receiver: '............1',
-            quant: amount,
+    setPermission(permission: Permission): Action {
+        return this.systemContract.action('updateauth', {
+            account: this.accountName,
+            auth: permission.required_auth,
+            authorized_by: '',
+            parent: permission.parent,
+            permission: permission.perm_name,
         })
     }
 
-    buyRamBytes(bytes: number): Action {
-        return this.eosioContract.action('buyrambytes', {
-            payer: '............1',
-            receiver: '............1',
-            bytes,
+    removePermission(permissionName: NameType): Action {
+        return this.systemContract.action('deleteauth', {
+            account: this.accountName,
+            authorized_by: '',
+            permission: permissionName,
         })
     }
 
-    sellRam(bytes: number): Action {
-        return this.eosioContract.action('sellram', {
-            account: '............1',
-            bytes,
-        })
+    linkauth() {
+        // TODO: Implement `linkauth` action calls
     }
 
-    delegateResources(cpu: AssetType, net: AssetType): Action {
-        return this.eosioContract.action('delegatebw', {
-            from: '............1',
-            receiver: '............1',
-            stake_cpu_quantity: cpu,
-            stake_net_quantity: net,
-            transfer: false,
-        })
+    unlinkauth() {
+        // TODO: Implement `unlinkauth` action calls
     }
 
-    undelegateResources(cpu: AssetType, net: AssetType): Action {
-        return this.eosioContract.action('undelegatebw', {
-            from: '............1',
-            receiver: '............1',
-            unstake_cpu_quantity: cpu,
-            unstake_net_quantity: net,
+    buyRam(amount: AssetType, options?: BuyramOptions): Action {
+        let receiver = this.accountName
+        if (options && options.receiver) {
+            receiver = Name.from(options.receiver)
+        }
+        return this.systemContract.action('buyram', {
+            payer: this.accountName,
+            quant: amount,
+            receiver,
         })
     }
 
-    getResources(): Resources {
```

Comment:
Don't you think that there is value in having a method that returns all resource data?